### PR TITLE
Screenshots #744 -> #613

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

Merge the #744 PR into the #613 PR to see what things are needed to change to merge the result types.

### Bugs Fixed ###

- Related to PR #613
- Related to PR #744
- Related to issue #125 

### Behavioral Changes ###

New screenshot API.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/Screenshot/Screenshot.android.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            var stream = new MemoryStream();
+
+            var f = format switch
+            {
+                ScreenshotFormat.Jpeg => Bitmap.CompressFormat.Jpeg,
+                _ => Bitmap.CompressFormat.Png,
+            };
+
+            await bmp.CompressAsync(f, 100, stream);
```
Comment (pictos) [inline Xamarin.Essentials/Screenshot/Screenshot.android.cs:62]: Can we add the `ConfigureAwait(false)` here?

File: Xamarin.Essentials/Screenshot/Screenshot.uwp.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            true;
+
+        static async Task<ScreenshotResult> PlatformCaptureAsync()
+        {
+            var element = Window.Current?.Content as FrameworkElement;
+            if (element == null)
+                throw new InvalidOperationException("Unable to find main window content.");
+
+            var bmp = new RenderTargetBitmap();
+            await bmp.RenderAsync(element);
```
Comment (pictos) [inline Xamarin.Essentials/Screenshot/Screenshot.uwp.cs:24]: same here, about `ConfigureAwait`

File: Xamarin.Essentials/Screenshot/Screenshot.android.cs (hunk 1 above)
Comment (mattleibow) [reply Xamarin.Essentials/Screenshot/Screenshot.android.cs]: We can!

File: Xamarin.Essentials/Screenshot/Screenshot.uwp.cs (hunk 2 above)
Comment (mattleibow) [reply Xamarin.Essentials/Screenshot/Screenshot.uwp.cs]: Done!

File: Xamarin.Essentials/Screenshot/Screenshot.shared.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        public static bool IsCaptureAvailable
+            => PlatformCanCapture;
```
Comment (mattleibow) [inline Xamarin.Essentials/Screenshot/Screenshot.shared.cs:9]: Not sure what we want to do here...

I see the media picker has a public property, but the app actions does not.

**Pull request merged**